From 61afe625564abcaf0e9cd21ac97ebc2a51a01062 Mon Sep 17 00:00:00 2001
From: Alexander Volkov <avolkov@astralinux.ru>
Date: Mon, 6 Mar 2023 18:10:34 +0300
Subject: [PATCH 125/144] Avoid crash when keysymToQtKey(keysym,
 Qt::ControlModifier) is called

For example when Ctrl+Home is sent by virtual keyboard on Wayland.

Pick-to: 6.6 6.5 6.2 5.15
Change-Id: I41f1d2a28c9091efa621d5826a3b9e3e0e481ceb
Reviewed-by: Liang Qi <liang.qi@qt.io>
(cherry picked from commit e68a0da0b907af7980481a3cf30f6b4b97c06b6e)
---
 src/platformsupport/input/xkbcommon/qxkbcommon.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/platformsupport/input/xkbcommon/qxkbcommon.cpp b/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
index 4d5336e1b3..ff556a0e60 100644
--- a/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
+++ b/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
@@ -755,6 +755,8 @@ xkb_keysym_t QXkbCommon::lookupLatinKeysym(xkb_state *state, xkb_keycode_t keyco
 {
     xkb_layout_index_t layout;
     xkb_keysym_t sym = XKB_KEY_NoSymbol;
+    if (!state)
+        return sym;
     xkb_keymap *keymap = xkb_state_get_keymap(state);
     const xkb_layout_index_t layoutCount = xkb_keymap_num_layouts_for_key(keymap, keycode);
     const xkb_layout_index_t currentLayout = xkb_state_key_get_layout(state, keycode);
-- 
2.43.0

